src/components/offerPage/index.jsx

```diff
@@ -0,0 +1,347 @@
+  });
+
+  let history = useHistory();
+
+  const [map, setMap] = useState();
+  const [clickedLatLng, setClickedLatLng] = useState(center);
+  const [data, setData] = useState(null);
+
+  const [address, setAddress] = useState();
+
+  const [form] = Form.useForm();
+
+  const handleSelect = async (value) => {
+    const results = await geocodeByAddress(value);
+    const latLng = await getLatLng(results[0]);
+    setAddress(value);
+    setClickedLatLng(latLng);
+  };
+
+  const onLoad = useCallback(function callback(map) {
+    const bounds = new window.google.maps.LatLngBounds(center);
+    map.fitBounds(bounds);
+    setMap(map);
+  }, []);
+
+  const onUnmount = useCallback(function callback(map) {
+    setMap(null);
+  }, []);
+
+  // get address and other from coordinates
+  const getData = (lat, lng) => {
+    Geocode.fromLatLng(lat, lng).then(
+      (response) => {
+        const address = response.results[0].formatted_address;
+        let settlement, region;
+        for (
+          let i = 0;
+          i < response.results[0].address_components.length;
+          i++
+        ) {
+          for (
+            let j = 0;
+            j < response.results[0].address_components[i].types.length;
+            j++
+          ) {
+            switch (response.results[0].address_components[i].types[j]) {
+              case "locality":
+                settlement =
+                  response.results[0].address_components[i].long_name;
+                break;
+              case "administrative_area_level_1":
+                region = response.results[0].address_components[i].long_name;
+                break;
+            }
+          }
+        }
+
+        form.setFieldsValue({
+          address: address,
+          settlement: settlement,
+          region: region,
+        });
+      },
+      (error) => {
+        console.error(error);
+      }
+    );
+  };
+
+  // for mapping good categories
+  useEffect(async () => {
+    setData(await getGoodCategories());
+  }, []);
+
+  // form actions
+  function disabledDate(current) {
+    // Can not select days before today
+    return current && current < moment().startOf("day");
+  }
+
+  const onFinish = (values) => {
+    createOffer(values, clickedLatLng, history);
+  };
+
+  const onFinishFailed = (values) => {
+    errorMessage(
+      offerErrorMessages.CREATE_OFFER_FAILED,
+      offerErrorMessages.ENTER_ALL_INPUTS
+    );
+  };
+
+  return isLoaded ? (
+    <>
+      <Header />
+      <div className="createOfferBody">
+        <h1>Create offer</h1>
+        <Form
+          form={form}
+          labelCol={{ span: 8 }}
+          wrapperCol={{ span: 16 }}
+          initialValues={{ remember: true }}
+          onFinish={onFinish}
+          onFinishFailed={onFinishFailed}
+          scrollToFirstError
+          className="formOffer"
+        >
+          <div className="leftFormBlock">
+            <Form.Item
+              name="address"
+              rules={[
```

Comment:
I can enter anything in this line, are you sure this is a good validation and that it matches the one in the service?